crates/control/src/kick_selector.rs

```diff
@@ -132,53 +146,209 @@ impl KickSelector {
+    let ball_near_left_opponent_corner =
+        distance(ball_position, left_opponent_corner) < parameters.distance_to_corner;
+    let ball_near_right_opponent_corner =
+        distance(ball_position, right_opponent_corner) < parameters.distance_to_corner;
+    ball_near_left_opponent_corner || ball_near_right_opponent_corner
+}
+
+fn determine_playing_situation(
+    filtered_game_controller_state: Option<&FilteredGameControllerState>,
+    ball_position: Point2<Field>,
+    field_dimensions: &FieldDimensions,
+    parameters: &DecisionParameters,
+) -> PlayingSituation {
+    let is_ball_in_opponent_corner =
+        is_ball_in_opponents_corners(ball_position, field_dimensions, parameters);
+    match filtered_game_controller_state {
+        Some(FilteredGameControllerState {
+            game_state: FilteredGameState::Playing { kick_off: true, .. },
+            game_phase: GamePhase::Normal,
+            opponent_game_state:
+                FilteredGameState::Playing {
+                    ball_is_free: false,
+                    ..
+                },
+            ..
+        }) => PlayingSituation::KickOff,
+        Some(FilteredGameControllerState {
+            game_phase: GamePhase::PenaltyShootout { .. },
+            kicking_team: Team::Hulks,
+            ..
+        })
+        | Some(FilteredGameControllerState {
+            sub_state: Some(SubState::PenaltyKick),
+            kicking_team: Team::Hulks,
+            ..
+        }) => PlayingSituation::PenaltyShot,
+        _ if is_ball_in_opponent_corner => PlayingSituation::CornerKick,
+        _ => PlayingSituation::Normal,
+    }
+}
+
+fn collect_kick_targets(
+    context: &CycleContext,
+    playing_situation: PlayingSituation,
+) -> Vec<Point2<Ground>> {
+    match playing_situation {
+        PlayingSituation::KickOff => generate_kick_off_kick_targets(context),
+        PlayingSituation::CornerKick => generate_corner_kick_targets(context),
+        PlayingSituation::PenaltyShot => generate_penalty_shot_kick_targets(context),
+        PlayingSituation::Normal => generate_goal_line_kick_targets(context),
+    }
+}
+
+fn generate_corner_kick_targets(context: &CycleContext) -> Vec<Point2<Ground>> {
+    let field_to_ground = context.ground_to_field.inverse();
+    let field_dimensions = &context.field_dimensions;
+    let parameters = &context.decision_parameters;
+
+    let from_corner_kick_target_x =
+        field_dimensions.length / 2.0 - parameters.corner_kick_target_distance_to_goal;
+    let target = field_to_ground * point![from_corner_kick_target_x, 0.0];
+    vec![target]
+}
+
+fn generate_goal_line_kick_targets(context: &CycleContext) -> Vec<Point2<Ground>> {
+    let field_to_ground = context.ground_to_field.inverse();
+    let field_dimensions = &context.field_dimensions;
+
+    let left_goal_half = field_to_ground
+        * point![
+            field_dimensions.length / 2.0 + 0.1,
```

Comment:
Can we make the additional offset dependable on distance to the goal? 
- Distance should be clamped to zero if we're inside the goal
- This could also allow us to still score goals if we're really delocalized in x

Reply:
Maybe next iteration. This is what @PatrickGoettsch decided to do for now. I did not change this in comparison to `main`